### PR TITLE
deal with no agent-install.crt, exit if docker is not installed

### PR DESCRIPTION
**Pull request by t-fine**

### CODE REVIEW

File: agent-install/agent-install.sh [hunk 1]
```diff
@@ -670,11 +670,7 @@ function install_linux(){
     if command -v docker >/dev/null 2>&1; then
         log_info "docker found"
     else
-        log_info "docker not found, installing it..."
-        set -x
-        apt install -y docker-ce docker-ce-cli containerd.io
-        { set +x; } 2>/dev/null
-        log_info "docker installed"
+        log_info "docker not found, install before continuing..."
```
Comment (linggao) [inline agent-install/agent-install.sh:673]: Should it exit here?

Comment (t-fine) [reply agent-install/agent-install.sh:673]: Ah shoot, yes, totally forgot the exit!

File: agent-install/agent-install.sh [hunk 2]
```diff
@@ -453,7 +453,7 @@ function create_config() {
 				-e "s~^HZN_DEVICE_ID=[^ ]*~HZN_DEVICE_ID=${DEVICE_ID}~g" \
 				-e "s~^HZN_FSS_CSSURL=[^ ]*~HZN_FSS_CSSURL=${HZN_FSS_CSSURL}~g" /etc/default/horizon
 		else
-			if [[ ${CERTIFICATE:0:1} != "/" ]]; then
+			if [[ ${CERTIFICATE:0:1} != "/" ]] && [[ -f "$CERTIFICATE" ]]; then
```
Comment (linggao) [inline agent-install/agent-install.sh:456]: Should change to
 ````
if [[ ${CERTIFICATE:0:1} != "/" ]] ; then 
     if [[ -f "$CERTIFICATE" ]]; then
	sudo mv $CERTIFICATE /etc/horizon/agent-install.crt
	ABS_CERTIFICATE=/etc/horizon/agent-install.crt
   fi
else
    ABS_CERTIFICATE=${CERTIFICATE}
fi
````

File: agent-install/agent-install.sh [hunk 3]
```diff
@@ -485,7 +485,7 @@ function install_macos() {
 
     sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain ${PACKAGES}/${MAC_PACKAGE_CERT}
     { set +x; } 2>/dev/null
-	if [[ "$CERTIFICATE" != "" ]]; then
+	if [[ "$CERTIFICATE" != "" ]] && [[ -f "$CERTIFICATE" ]]; then
```
Comment (linggao) [inline agent-install/agent-install.sh:488]: Should changed to 
````
if [[ "$CERTIFICATE" != "" ]]; then
    log_info "Configuring an edge node to trust the ICP certificate ..."
    set -x
    if [[ -f "$CERTIFICATE" ]]; then
		sudo mv $CERTIFICATE /private/etc/horizon/agent-install.crt
    fi
    CERTIFICATE=/private/etc/horizon/agent-install.crt
    sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain "$CERTIFICATE"
    { set +x; } 2>/dev/null
fi

File: agent-install/agent-install.sh [hunk 4]
```diff
@@ -584,7 +584,7 @@ function install_macos() {
 		printf "HZN_EXCHANGE_URL=${HZN_EXCHANGE_URL} \nHZN_FSS_CSSURL=${HZN_FSS_CSSURL} \
 			\nHZN_DEVICE_ID=${DEVICE_ID}"  | sudo tee "$HZN_CONFIG"
 	else
-		if [[ ${CERTIFICATE:0:1} != "/" ]]; then
+		if [[ ${CERTIFICATE:0:1} != "/" ]] && [[ -f "$CERTIFICATE" ]]; then
```
Comment (linggao) [inline agent-install/agent-install.sh:587]: should change to 
````
if [[ ${CERTIFICATE:0:1} != "/" ]]; then
    [[ -f "$CERTIFICATE" ]]; then
             sudo mv $CERTIFICATE /private/etc/horizon/agent-install.crt
    fi
    ABS_CERTIFICATE=/private/etc/horizon/agent-install.crt
else
    ABS_CERTIFICATE=${CERTIFICATE}
fi

**Pull request merged**